include/boost/geometry/algorithms/detail/overlay/get_turn_info_helpers.hpp

```diff
@@ -278,34 +404,49 @@ class intersection_info
     // TODO: it's more like is_spike_ip_p
     inline bool is_spike_p() const
     {
+        if (base::m_range_p.size() == 2u)
```

Comment:
Out of curiosity, in which case something like this can happen? Is this a check for invalid input or a check for last segment as it was before?